File: src/main/java/mil/nga/tiff/field/type/enumeration/Threshholding.java
package mil.nga.tiff.field.type.enumeration;

import mil.nga.tiff.util.TiffException;

import java.util.Arrays;

public enum Threshholding {
    NO(1),
    ORDERED(2),
    RANDOM(3);

    private final int id;

    Threshholding(int id) {
        this.id = id;
    }

    public static Threshholding findById(int id) {
        return Arrays.stream(values())
            .filter(o -> o.id == id)
            .findAny()
            .orElseThrow(() -> new TiffException("Invalid ID: " + id));
    }

    public int getId() {
        return id;
    }
}

File: src/main/java/mil/nga/tiff/field/type/enumeration/Threshholding.kt
package mil.nga.tiff.field.type.enumeration

enum class Threshholding(val id: Int) {
    NO(1),
    ORDERED(2),
    RANDOM(3);

    companion object {
        fun findById(id: Int): Threshholding {
            return entries.first { it.id == id}
        }
    }
}
